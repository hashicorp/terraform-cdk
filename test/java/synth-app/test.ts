// Copyright (c) HashiCorp, Inc
// SPDX-License-Identifier: MPL-2.0
import { TestDriver, onlyHcl, onlyJson } from "../../test-helper";

describe("java full integration", () => {
  let driver: TestDriver;

  beforeAll(async () => {
    driver = new TestDriver(__dirname);
    await driver.setupJavaProject();
  });

  test("debug command", async () => {
    driver.setEnv("CDKTF_LOG_LEVEL", "debug");
    await driver.exec(`cdktf debug --json`);
    driver.setEnv("CDKTF_LOG_LEVEL", "warning");
    const { stdout } = await driver.exec(`cdktf debug --json`);
    const { cdktf, constructs } = JSON.parse(stdout);
    expect(cdktf.length).not.toBe(0);
    expect(constructs.length).not.toBe(0);
  });

  onlyJson("synth generates JSON", async () => {
    await driver.synth();
    expect(driver.synthesizedStack("java-simple").toString()).toMatchSnapshot();
  });

  onlyHcl("synth generates HCL", async () => {
    await driver.synth();
    expect(
      driver.synthesizedStackContentsRaw("java-simple").toString()
    ).toMatchSnapshot();
  });
});
