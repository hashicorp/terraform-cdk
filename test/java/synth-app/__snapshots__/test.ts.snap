// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`java full integration synth generates HCL 1`] = `
"terraform {
  required_providers {
    null = {
      version = "3.1.1"
      source  = "hashicorp/null"
    }
    random = {
      version = "3.1.3"
      source  = "hashicorp/random"
    }
  }
  backend "local" {
    path = "terraform.tfstate"
  }


}

provider "null" {
}
resource "null_resource" "NullResource" {
}

provider "random" {
}
resource "random_string" "RandomString" {
  length = 42
}"
`;

exports[`java full integration synth generates JSON 1`] = `
"{
  "//": {
    "metadata": {
      "backend": "local",
      "stackName": "java-simple",
      "version": "stubbed"
    },
    "outputs": {}
  },
  "provider": {
    "null": [
      {}
    ],
    "random": [
      {}
    ]
  },
  "resource": {
    "null_resource": {
      "NullResource": {
        "//": {
          "metadata": {
            "path": "java-simple/NullResource",
            "uniqueId": "NullResource"
          }
        }
      }
    },
    "random_string": {
      "RandomString": {
        "//": {
          "metadata": {
            "path": "java-simple/RandomString",
            "uniqueId": "RandomString"
          }
        },
        "length": 42
      }
    }
  },
  "terraform": {
    "backend": {
      "local": {
        "path": "terraform.tfstate"
      }
    },
    "required_providers": {
      "null": {
        "source": "hashicorp/null",
        "version": "3.1.1"
      },
      "random": {
        "source": "hashicorp/random",
        "version": "3.1.3"
      }
    }
  }
}"
`;
